client/python/lib/test/test_tensors.py

```diff
@@ -104,3 +104,135 @@ def test_make_ndarray_invalid_no_type():
+    _shape = TensorShapeProto(dim = [TensorShapeProto.Dim(size=1), TensorShapeProto.Dim(size=2), TensorShapeProto.Dim(size=3)])
+    tensor_proto = TensorProto(tensor_shape=_shape, dtype = DataType.DT_FLOAT, tensor_content = np_values.tobytes())
+    array = make_ndarray(tensor_proto)
+    tensor_proto = make_tensor_proto(values = array, dtype = DataType.DT_FLOAT)
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.dtype == DataType.DT_FLOAT
+    assert tensor_proto.tensor_shape == _shape
+
+def test_make_tensor_proto_valid_no_dtype_provided():
+    values = [1.0, 2.0, 3.0]
+    tensor_proto = make_tensor_proto(shape=[3], values = values)
+    np_values = np.array(values)
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.dtype == DataType.DT_DOUBLE
+    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=3)])
+
+def test_make_tensor_proto_valid_no_shape_provided():
+    values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
+    tensor_proto = make_tensor_proto(dtype = DataType.DT_FLOAT, values = values)
+    expected_shape = TensorShapeProto(dim = [TensorShapeProto.Dim(size=2), TensorShapeProto.Dim(size=3)])
+    np_values = np.array(values, np.float32)
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.tensor_shape == expected_shape
+    assert tensor_proto.dtype == DataType.DT_FLOAT
+
+def test_make_tensor_proto_invalid_dtype_provided():
+    values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
+    with pytest.raises(TypeError) as exception_info:
+        make_tensor_proto(shape = None, dtype = DataType.DT_STRING, values = values)
+
+    exception = exception_info.value
+    assert str(exception) == "[1.0, 2.0, 3.0] has type list, but expected one of: bytes"
+
+def test_make_tensor_proto_invalid_dimsions():
+    values = [[1.0, 2.0], [1.0, 2.0, 3.0]]
+    with pytest.raises(ValueError) as exception_info:
+        make_tensor_proto(shape=[2,3], dtype = DataType.DT_FLOAT, values = values)
+
+    exception = exception_info.value
+    assert str(exception) == "shape provided does not match values array shape"
+
+def test_make_tensor_proto_invalid_shape_type():
+    values = 5.0
+    with pytest.raises(TypeError) as exception_info:
+        make_tensor_proto(shape=0, dtype = DataType.DT_FLOAT, values = values)
+
+    exception = exception_info.value
+    assert str(exception) == "shape type should be list, but is int"
+
+def test_make_tensor_proto_invalid_dimensions_no_shape_provided():
+    values = [[1,2,3], [4,5]]
+    with pytest.raises(ValueError) as exception_info:
+        make_tensor_proto(shape=None, dtype = DataType.DT_INT8, values = values)
+
+    exception = exception_info.value
+    assert str(exception) == "setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part."
```

Comment:
In make_tensor_proto with shape=None, we determine the shape ourselves and we don't check whether it matches values list. Would it be better to check
```
list(np.array(values).shape) == infered_shape
```
and raise an exception if they do not match @dkalinowski ?
In this test the comparsion would be: [2] == [2,3]

Reply:
I think this test would fail for other reason. `[[1,2,3], [4,5]]` is not valid numpy array due to number of elements in each row.
But to answer your question I think it's not neccesssary check since we use numpy inside make_tensor_proto to get the shape anyway.

Reply:
Can we also bring this part under `with` scope?